storage/innobase/row/row0ins.cc

```diff
@@ -1806,13 +1806,20 @@ row_ins_check_foreign_constraint(
 				if (check_ref) {
 					err = DB_SUCCESS;
 #ifdef WITH_WSREP
+					Wsrep_key_type key_type;
+					if (upd_node != NULL &&
+					    wsrep_protocol_version < 4) {
+						key_type = wsrep_key_shared;
+					} else {
+						key_type = wsrep_key_reference;
+					}
 					err = wsrep_append_foreign_key(
 						thr_get_trx(thr),
 						foreign,
 						rec,
 						check_index,
 						check_ref,
-						upd_node != NULL ? WSREP_KEY_SHARED : WSREP_KEY_SEMI);
+						key_type);
```

Comment:
I liked the original code better, because it did not introduce an initially uninitalized local variable for something that is only used once.